src/components/ChatApp/MessageComposer.react.js

```diff
@@ -47,31 +41,43 @@ class MessageComposer extends Component {
       stop: false,
       open: false,
       result:'',
+      animate: false
     };
     if(props.dream!==''){
       this.state= {text: 'dream '+ props.dream}
     }
   }
 
   onStart = () => {
-    this.setState({ start: true, stop: false, open:true})
+    this.setState({ start: true, stop: false, open:true,
+      animate:true})
   }
 
   onEnd = () => {
-    this.setState({ start: false, stop: false, open: false})
+    this.setState({ start: false, stop: false, open: false,
+      animate:false})
   }
 
-  onResult = ({finalTranscript }) => {
-    let result = finalTranscript;
-    this.setState({ result:result});
-    setTimeout(()=>this.setState({ start: false, stop: false, open:false}),400);
+  onResult = ({interimTranscript,finalTranscript }) => {
+    let result = interimTranscript;
     let voiceResponse = false;
-    if(this.props.speechOutputAlways || this.props.speechOutput){
-      voiceResponse = true;
+    this.setState({result:result});
+    if(finalTranscript) {
+      result = finalTranscript;
+      this.setState({
+      start: false,
+      result:result,
+      stop: false,
+      open:false,
+      animate:false
+      });
+      Actions.createMessage(result, this.props.threadID, voiceResponse);
```

Comment:
Looks good :+1: 
Just one thing, voice output for voice input is not working. Please check once.
I think this needs to be :

if(this.props.speechOutputAlways || this.props.speechOutput){
  voiceResponse = true;
}
Actions.createMessage(result, this.props.threadID, voiceResponse);